🐛 Fixed post scheduling for sites with transferred ownership

cmraible opened the pull request:
refs TryGhost/Team#3494

- By default, the post scheduler runs as user_id = 1, which is the original owner of the site
- It is possible to transfer ownership to another user, then delete the original owner, so some sites don't have a user_id = 1
- With the addition of post history, we save the author_id from the contextUser, which is usually the logged in user
- However when the scheduler runs and tries to publish a post, it sets the author_id to 1, which then fails the foreign key constraint on the post_revisions table
- This commit fixes the issue by checking if the contextUser exists, and falling back to the author of the most recent post_revision if it doesn't
- This is a bit of a hack/band-aid to fix the immediate issue, but ideally this check would be made in the contextUser function, so we can always be confident the contextUser exists without firing an ad-hoc DB query
- Additionally, we should refactor to query for the current owner's user_id instead of defaulting to 1

renovate[bot] committed:
Update dependency tough-cookie to v4.1.3

renovate[bot] committed:
Update dependency @extractus/oembed-extractor to v3.1.10

renovate[bot] committed:
Update dependency gscan to v4.37.0

rshbhgrg committed:
Released Admin-x-settings v0.0.8

renovate[bot] committed:
Update storybook monorepo to v7.0.22

binary-koan committed:
Added errors and improved behaviour of the navigation editor

refs https://github.com/TryGhost/Team/issues/3432

daniellockyer commented on ghost/core/core/server/models/post.js:
```diff
@@ -875,7 +876,8 @@ Post = ghostBookshelf.Model.extend({
             });
-            const authorId = this.contextUser(options);
+            let authorId = this.contextUser(options);
+            const authorExists = User.findOne({id: authorId});
```
I think we're missing an await here

ErminMerdanovic committed:
Updated German translations (#17045)

refs https://github.com/TryGhost/Team/issues/2795

daniellockyer committed:
Moved signup-form to `apps/` folder

refs https://github.com/TryGhost/Toolbox/issues/594

- we're moving all the standalone apps to a separate folder to keep them
  out of the core code and easier to find

naz committed:
Added basic UI for collection posts

refs https://github.com/TryGhost/Team/issues/3169

- Added a lits to a collection view showing which posts belong to a collection.

allouis reviewed (commented):
I think using the previous revisions author is weird because it's going to provide different behaviour dependent on the state of the site 

What about updating the `this.contextUser` method to:
1. Return a Promise
2. Return the Owner User id rather than `1`

🤔

naz committed:
Fixed slug-based navigation in collections UI

refs https://github.com/TryGhost/Team/issues/3169

- When navigating through collections slugs did not display correctly returning "undefined" in the navigation. It was easier to fix the bug than working around it.

kevinansfield committed:
🐛 Fixed beta editor HTML cards auto-closing tags when rendering

closes https://github.com/TryGhost/Team/issues/3462
refs https://github.com/TryGhost/Koenig/commit/a49f296c08a6fbc84b1f399989a21c5fd23400d5

- bumps `@tryghost/kg-default-nodes` and `@tryghost/kg-lexical-html-renderer` with support for "raw" html values in rendered card output

honhatduy committed:
Updated Vietnamese translations (#17052)

refs https://github.com/TryGhost/Team/issues/2795

honhatduy committed:
Updated Vietnamese translations (#17051)

refs https://github.com/TryGhost/Team/issues/2795

kevinansfield committed:
Fixed "Beta feedback" not showing in beta editor when feature flag is switched off

no issue

- even when the editor beta is turned off, any post created in the beta will still open in the beta editor
- the "beta feedback" text in the bottom right was disabled when the beta flag was disabled which meant it was difficult to tell when you were using the beta editor vs the original editor

renovate[bot] committed:
Update dependency @vitejs/plugin-react to v4.0.1

binary-koan committed:
Updated URL editing to not automatically force everything to URLs

refs https://github.com/TryGhost/Team/issues/3433

cmraible commented:
> I think using the previous revisions author is weird because it's going to provide different behaviour dependent on the state of the site
> 
> What about updating the `this.contextUser` method to:
> 
> 1. Return a Promise
> 2. Return the Owner User id rather than `1`
> 
> 🤔

Agree with you that this would be a better solution, but I'm struggling to implement it properly...I just pushed what I have right now to this branch — this should work in theory, but it breaks a bunch of tests. I think the pattern of relying on '1' being the owner user is still pretty entrenched in the codebase, especially in the test framework. 

I'm having a hard time untangling the circular dependency on the owner user — e.g. when we create the owner user in the fixture-utils, we rely on the owner user already existing https://github.com/TryGhost/Ghost/blob/82acf85b297279639a3c14a7b215d8e6d9c0ea2c/ghost/core/test/utils/fixture-utils.js#L225

Let me know your thoughts, struggling with this one a bit.

binary-koan committed:
Added dirty state navigation blocking to AdminX settings

refs https://github.com/TryGhost/Team/issues/3349

daniellockyer committed:
Merged v5.52.1 into main

honhatduy committed:
Updated Vietnamese translations (#17053)

refs https://github.com/TryGhost/Team/issues/2795

truekasun committed:
Updated Sinhalese translations (#17062)

refs https://github.com/TryGhost/Team/issues/2795

truekasun committed:
Updated Sinhalese translations (#17063)

refs https://github.com/TryGhost/Team/issues/2795

binary-koan committed:
Updated add navigation button to validate the new item

refs https://github.com/TryGhost/Team/issues/3433

binary-koan committed:
Improved naming in AdminX dirty state handling

refs https://github.com/TryGhost/Team/issues/3349

truekasun committed:
Updated Sinhalese translations (#17060)

refs https://github.com/TryGhost/Team/issues/2795

truekasun committed:
Updated Sinhalese translations (#17064)

refs https://github.com/TryGhost/Team/issues/2795

yovko committed:
Updated Bulgarian translations (#16991)

refs https://github.com/TryGhost/Team/issues/2795

renovate[bot] committed:
Update dependency socket.io to v4.6.2

renovate[bot] committed:
Update dependency @sinonjs/fake-timers to v11

renovate[bot] committed:
Update dependency @types/jsonwebtoken to v9

Commit:
🎨 Improved copy on too many login attempts

fixes https://github.com/TryGhost/Team/issues/2391

daniellockyer committed:
Updated translation files

- contains some changes produced by i18next-parser

daniellockyer committed:
Updated vitest to v0.32

refs https://github.com/TryGhost/Ghost/pull/16949
refs https://github.com/vitest-dev/vitest/releases/tag/v0.32.0

- contains a switch from `@vitest/coverage-c8` to `@vitest/coverage-v8`
  as this is now required in 0.32

renovate[bot] committed:
Update dependency luxon to v3

ronaldlangeveld committed:
🎨 Improved copy on page & post revisions (#17068)

no issue

Updates the labels of the post history components to reflect the type of
content (post or page).
It modifies the `restore-revision`, `gh-post-settings-menu`, and
`modal-post-history` components.

renovate[bot] committed:
Update metascraper to v5.34.7

cmraible committed:
🐛 Fixed scheduled posts not publishing in beta editor if ownership has been transferred

refs @TryGhost/Team#3494

- Posts created in the beta editor and scheduled to publish were not publishing successfully if ownership of the site had been transferred to another user
- The scheduler runs as an 'internal' user, which defaults to the user with id = 1
- If ownership of the site has been transferred, the user with id = 1 may not exist, and the scheduler was unable to publish posts due to a foreign key constraint failing
- This commit fixes the issue by checking if the contextUser exists, and replacing it with the owner user if it does not, before updating the post
